extensions/resteasy-classic/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/standalone/ResteasyConfigurationMPConfig.java

```diff
@@ -67,6 +67,11 @@ public Set<String> getInitParameterNames() {
     }
 
     private static Optional<String> getGzipMaxInput(Config config) {
+        if (config.getOptionalValue("resteasy.gzip.max.input", String.class).isPresent()) {
+            // resteasy-specific properties have priority
+            return Optional.empty();
+        }
```

Comment:
Mkay. We can do that if it doesn't cost us anything but I don't see it as mandatory.

You're returning `Optional.empty()` though, is it expected?

Reply:
This is how I understand how it works:

- resteasy specific configuration properties are propagated OOTB, because:
  - `GreetingResourceTest` passes in the current main branch and it requests `resteasy.gzip.max.input`
  - setting `quarkus.resteasy.gzip.max.input` without condition you are commenting on overrides `resteasy.gzip.max.input` and `GreetingResourceTest`, therefore we know it works and RESTEasy has priority